src/WinSparkleDotNet/WinSparkle.cs

```diff
@@ -0,0 +1,557 @@
+        private static extern int GetUpdateCheckInterval();
+
+        /// <summary>   
+        ///     <para>
+        ///         Gets the time for the last update check.
+        ///     </para>
+        ///     <para>
+        ///         Default value is -1, indicating that the update check has never run.
+        ///     </para>
+        /// </summary>
+        ///
+        /// <remarks>
+        ///     <para>
+        ///         Can only be called @em before the first call to <see cref="Initialize"/>
+        ///     </para>
+        /// </remarks>
+        ///
+        /// <returns>   The last check time internal. </returns>
+        [DllImport("WinSparkle.dll", EntryPoint = "win_sparkle_get_last_check_time", CallingConvention = CallingConvention.Cdecl)]
+        [return: MarshalAs(UnmanagedType.I4)]
+        private static extern int GetLastCheckTime();
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        public delegate void ErrorCallback();
+
+        /// <summary>   
+        ///     <para>
+        ///         Set callback to be called when the updater encounters an error.
+        ///     </para>
+        /// </summary>
+        /// <param name="callback"> The callback. </param>
+        [DllImport("WinSparkle.dll", EntryPoint = "win_sparkle_set_error_callback", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
+        public static extern void SetErrorCallback(
+            [param: MarshalAs(UnmanagedType.FunctionPtr)] ErrorCallback callback
+            );
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        public delegate bool CanShutdownCallback();
+
+        /// <summary>   
+        ///     <para>
+        ///         Set callback for querying the application if it can be closed.
+        ///     </para>
+        ///     <para>
+        ///         This callback will be called to ask the host if it's ready to shut down,
+        ///         before attempting to launch the installer. The callback returns TRUE if
+        ///         the host application can be safely shut down or FALSE if not (e.g. because
+        ///         the user has unsaved documents).
+        ///     </para>
+        /// </summary>
+        ///
+        /// <remarks>
+        ///     <para>
+        ///         Can only be called @em before the first call to <see cref="Initialize"/>
+        ///     </para>
+        ///     <para>
+        ///         There's no guarantee about the thread from which the callback is called,
+        ///         except that it certainly *won't* be called from the app's main thread.
+        ///         Make sure the callback is thread-safe.
+        ///     </para>
+        /// </remarks>
+        ///
+        /// <param name="callback"> The callback. </param>
+        [DllImport("WinSparkle.dll", EntryPoint = "win_sparkle_set_can_shutdown_callback", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
```

Comment:
Again, no need to specify the `CharSet = CharSet.Ansi` (and there are more places where this is not necessary).